src/ui/component/publishForm/index.js

```diff
@@ -34,7 +33,7 @@ const perform = dispatch => ({
   updatePublishForm: value => dispatch(doUpdatePublishForm(value)),
   clearPublish: () => dispatch(doClearPublish()),
   resolveUri: uri => dispatch(doResolveUri(uri)),
-  publish: params => dispatch(doPublish(params)),
+  publish: (success, fail) => dispatch(doPublish(success, fail)),
```

Comment:
We don't need to pass in the callbacks here, lets do it in redux (in the desktop codebase).